src/sql/magic_cmd.py

```diff
@@ -149,11 +165,18 @@ def execute(self, cmd_name="", others="", cell="", local_ns=None):
             conn = sql.connection.Connection.current.session
             result_dict = run_each_individually(args, conn)
 
-            if len(result_dict.keys()):
-                print(
-                    "Test failed. Returned are samples of the failures from your data:"
+            if any(len(rows) > 1 for rows in list(result_dict.values())):
+                for comparator, rows in result_dict.items():
+                    if len(rows) > 1:
+                        print(f"\n{comparator}:\n")
+                        _pretty = PrettyTable()
+                        _pretty.field_names = rows[0]
+                        for row in rows[1:]:
+                            _pretty.add_row(row)
+                        print(_pretty)
+                raise UsageError(
```

Comment:
If we raise any other type of error (e.g., `ValueError`), Jupyter will display a full traceback, which is pretty bad for UX since the user doesn't care about the internal traceback. If we raise `UsageError`, Jupyter only displays the error message. I agree `UsageError` isn't semantically correct but it's better than having the full traceback.

opened: https://github.com/ploomber/jupysql/issues/407